cmake/GNU.cmake

```diff
@@ -0,0 +1,68 @@
+if (CMAKE_Fortran_COMPILER_VERSION VERSION_LESS 5.3)
+  message(FATAL_ERROR "${CMAKE_Fortran_COMPILER_ID} version must be at least 5.3!")
+endif()
+
+set (FOPT0 "-O0")
+set (FOPT1 "-O1")
+set (FOPT2 "-O2")
+set (FOPT3 "-O3")
+set (FOPT4 "-O3")
+set (AVX2 "-O3")
```

Comment:
``AVX2`` variable should enable the use of AVX2 instructions (``-maxv2``), and should not set optimization level.